climada/hazard/tc_clim_change.py

```diff
@@ -1,174 +1,329 @@
-"""
-This file is part of CLIMADA.
+#
+# MIT License
+#
+# Copyright (c) 2021 Lynne Jewson, Stephen Jewson
+# Copyright (c) 2023 ETH Zurich, CLIMADA contributors listed in AUTHORS
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
 
-Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+from statistics import mean
+from math import log, exp
+import pandas as pd
+import numpy as np
 
-CLIMADA is free software: you can redistribute it and/or modify it under the
-terms of the GNU General Public License as published by the Free
-Software Foundation, version 3.
+MAP_BASINS_NAMES = dict(
+    zip(['NA', 'WP', 'EP', 'NI', 'SI', 'SP'], range(6))
+    )
 
-CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
-WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
-PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+MAP_VARS_NAMES = dict(
+    zip(['cat05', 'cat45', 'intensity'], range(3))
+    )
 
-You should have received a copy of the GNU General Public License along
-with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+def get_knutson_scaling_factor(
```

Comment:
Is my understanding correct that this function returns the scaling factors every 5 years from 2000 to 2100?
Why not rather return only the factors for one target year here?